pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -387,7 +387,10 @@ func (cb *ClusterBuilder) buildDefaultCluster(name string, discoveryType cluster
 			if features.EnableDualStack {
 				// If dual-stack, it may be [IPv4, IPv6] or [IPv6, IPv4]
 				// using Cluster_ALL to enable Happy Eyeballsfor upstream connections
-				c.DnsLookupFamily = cluster.Cluster_ALL
+				ipFamily := networkutil.CheckIPFamilyTypeForFirstIPs(cb.proxyIPAddresses)
+				if ipFamily == networkutil.IPv6 {
+					c.DnsLookupFamily = cluster.Cluster_AUTO
+				}
```

Comment:
else?

Reply:
why is V4_ONLY the value for IPv4-first DS clusters and not V4_PREFERRED?

Reply:
Hi @LithiumSR, thanks for your comments, `Cluster_V4_PREFERRED` would be better, I have changed it.

Reply:
Hi @ymesika, still using `else`  branch to make the same style as it in `pkg/bootstrap/config.go`

Reply:
Why not ALL for both?

Reply:
The linked issue doesn't seem to explain this. Why would DNS resolution be different if we are "ipv6 first" or "ipv4 first"? They are all just "dual stack", no?

Reply:
Hi @howardjohn, Basically, I agree with you that there should be okay for `Cluster_ALL` for it. I doubt that there may be some misconfiguration in `pkg/bootstrap/config.go`, let me try to make here the same as it in `pkg/bootstrap/config.go`.

Reply:
Hi @howardjohn,  it can be `Cluster_ALL` for both here, and it must make sure that which is the first for the logic in `pkg/bootstrap/config.go` even if I have tried all 4 situations. I will ask @jacob-delgado and @LithiumSR for help in this.